### DISCUSSION TITLE
change auto-suggest key behavior

### DISCUSSION THREAD

**Discussion by Kobaltauge** [chrisant996/clink#691] [Ideas]
Hi.
I read in [this post](https://github.com/chrisant996/clink/discussions/477#discussioncomment-6722395) that the auto-suggestion behavior is like in the fish shell. But I don't like it. Clicking ob the right arrow should have the behavior like "shift+right" and only with a modifier key it should accept the whole line. So I would like to switch the beaviour between "right" and "shift+right".

If this could be done by a setting, this would be awesome.

**Comment by chrisant996** [comment]
Customize your key bindings to your liking. 

Refer to the docs on how to customize key bindings.

**Reply by Kobaltauge** [reply]
Thank you. But it seems that I'm too dumb to understand it. Tried all this and nothing works:

this throws an error (as probaly expected)
Right   : cua-forward-char
S-Right  : win-cursor-forward

this blocks right and shift-right
"\e[C"   : cua-forward-char
"\e[1;2C"  : win-cursor-forward

this blocks shift-right and right behaves standard
"\e[C": forward-char
"\e[1;2C" : win-cursor-forward 

I know this is not the help section. I will continue with my tests.

**Reply by chrisant996** [reply]
Oh, I'm sorry, I was half asleep earlier and didn't think it through completely. 

You will need to write a custom command (it's a very simple one, don't worry). 

[This comment](https://github.com/chrisant996/clink/issues/637#issuecomment-2253812966) is very similar to what you'll need to write.  But instead you'll want to switch between your choice of [these commands](https://chrisant996.github.io/clink/clink.html#rlcmd-clink-insert-suggested-full-word) and [this command](https://chrisant996.github.io/clink/clink.html#rlcmd-cua-forward-char).

(And yes, the first way you tried is by design that it doesn't work -- but that's because Clink is using the Readline library which it's from Unix where there's not really a good way to identify a key by a human-readable name.  I would like to be able to use the human-readable key names there, but it would break compatibility with bash, so I haven't gone there.)

**Reply by chrisant996** [reply]
I don't want to add the kind of configuration setting you suggested, because it has messy side effects (like complicating documentation and breaking other Lua scripts, and more).

It's also a slippery slope, and can quickly devolve into requests for several more settings and commands.

That level of customization is best done in Lua script, like in the comment I linked in my previous reply.

**Reply by Kobaltauge** [reply]
Thank you so much for your help. And of course thx for your application. Love it!

**Reply by chrisant996** [reply]
> Thank you so much for your help. And of course thx for your application. Love it!

Did you get it working the way you want? 
Let me know if not.
